src/blake2b_functions.c

```diff
@@ -1,8 +1,192 @@
-#include <stdio.h>
 #include "blake2b.h"
+#include "utils.h"
+#include <math.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+/**
+ * @brief  The Mix function is called by the Compress function, and mixes two
+ *         8-byte words from the message into the hash state
+ *
+ * @param   v           the work vector V
+ * @params  a, b, c, d  indices to 8-byte word entries from the work vector V
+ * @params  x, y        two 8-byte word entries from padded message v
+ */
+static void
+G(uint64_t v[16], int a, int b, int c, int d, int64_t x, int64_t y)
```

Comment:
Will do. Would you want us to have some kind of type safety with the macro? 
Didn't know that bit about windows. Will look into that.